modules/collection/programs/nushell.nix

```
@@ -0,0 +1,203 @@
+      example = {
+        ll = "ls -l";
+        spp = "spotify_player";
+      };
+      description = ''
+        A set of aliases for nushell, converted into nu and written to
+        {file}`$HOME/.config/nushell/config.nu`. Please note that we cannot
+        handle complex aliases that use `def` at this time.
+
+        Please see [The Nushell Book] for a simple tutorial.
+
+        [The Nushell Book]: https://www.nushell.sh/book/aliases.html
+      '';
+    };
+
+    plugins = mkOption {
+      type = listOf package;
+      default = [];
+      example = literalExpression ''
+        with pkgs.nushellPlugins; [
+          units
+          formats
+          query
+        ]
+      '';
+      description = ''
+        A list of plugin packages to be installed and added to the
+        nushell plugin registry.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = lines;
+      default = "";
+      example = ''
+        def webdev [--run (-r)] {
+          cd ~/my-website
+          if $run {
+            nix develop --command pnpm run dev
+          } else {
+            nix develop --command nu
+          }
+        }
+      '';
+      description = ''
+        Extra configuration to be written to {file}`$HOME/.config/nushell/config.nu`.
+
+        Please see [The Nushell Book] for more info.
+
+        [The Nushell Book]: https://www.nushell.sh/book/configuration.html
+      '';
+    };
+
+    envFile = mkOption {
+      type = lines;
+      default = "";
+      description = ''
+        [The Nushell Book]: https://www.nushell.sh/book/configuration.html
+
+        Extra configuration to be written to {file}`$HOME/.config/nushell/env.nu`.
+        Please keep in mind that the upstream documentation generally advises against
+        writing to this file, and instead suggests using {file}`$HOME/.config/nushell/config.nu`.
+
+        See [The Nushell Book] for more info.
+
+      '';
+    };
+
+    loginFile = mkOption {
+      type = lines;
+      default = "";
+      description = ''
+        Nushell may be used as a login shell, so this option writes to
+        {file}`$HOME/.config/nushell/login.nu` to allow you to configure
+        nushell as your login shell.
+
+        Please recognize that this is mainly for advanced users and is not
+        for the faint of heart. Direct your questions and concerns to
+        [The Nushell Book] for more information.
+
+        [The Nushell Book]: https://www.nushell.sh/book/configuration.html#configuring-nu-as-a-login-shell
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    packages = [cfg.package];
+    files = let
+      checks = {
+        settings = cfg.settings != {};
+        aliases = cfg.aliases != {};
+        extraConfig = cfg.extraConfig != "";
+        variables = config.environment.sessionVariables != {};
+      };
+    in {
+      ".config/nushell/config.nu".text = mkIf (checks.settings || checks.aliases || checks.extraConfig || checks.variables) (
```

```suggestion
      ".config/nushell/config.nu".text = mkIf (checks.settings || checks.aliases || checks.extraConfig || checks.variables) (
```
```suggestion
      ".config/nushell/config.nu".text = mkIf (any id (attrValues checks)) (
```

Nit: this makes it more concise (taken from the zsh module)